features/step_definitions/operators.rb

```diff
@@ -160,3 +160,35 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to check conditions: "([^"]*)"$/ do | time, operator, conditions |
+  ensure_admin_tagged
+
+  interval_time = 15
+  expected = {}
+  # Parse the conditions to Hash value {Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}
+  if conditions.include? "="
+    conditions.split(/\s+/).map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
+  else
+    arr_conditions = conditions.split(/\s+/)
+    ["Available", "Progressing", "Degraded", "Upgradeable"][0,arr_conditions.length].zip(arr_conditions).map {|k, v| expected[k]=v}
+  end
+  raise "### Without valid conditions, unable to continue to check !!!" unless expected.length >0
```

Comment:
If you like, you could validate all values must be True or False.

Reply:
Line 170~173 I add validations as below,
```
if conditions.split(/\s+/).any? { |i| /.*=True|False/ } 
    arr_conditions = conditions.split(/\s+/)
    raise "### Invalid number of conditions!!! (Valid number:1~4)" unless arr_conditions.length <=4
    arr_conditions.split(/\s+/).map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
  else
    arr_conditions = conditions.split(/\s+/)
    raise "### Invalid conditions #{arr_conditions} !!! " unless arr_conditions.any? { |i| ["True","Flase"].include? i }
```

Reply:
`any?` means if any one satisfies. But we want all must satisfy. So `any?` isn't OK.
You can use
expected.each {|k, v| raise ... unless ["True","Flase"].include? v}

Reply:
sure, need to check each value